doc/specs/#653 - Quake Mode/#653 - Quake Mode.md

```diff
@@ -0,0 +1,661 @@
+* `monitor`
+  - `"any"`: Leave the window on the monitor it is already on.
+  - `"toCurrent"`: If the window is on another monitor, then move it to the
+    currently active one.
+  - `"onCurrent"`: If the window is on another monitor, then move it to the
+    currently active one.
+  - `<int>`: If the window is on another monitor, then move it to the specified
+    monitor.
+
+> NOTE: You read that right, `onCurrent` and `toCurrent` both do the same thing
+> when `window` is provided. They both already know which window to select, the
+> context of moving to the "current" monitor is all that those parameters add.
+
+#### Other properties
+
+Some users would like the terminal to just appear when the global hotkey is
+pressed. Others would like the true quake-like experience, where the terminal
+window "slides-in" from the top of the monitor. Furthermore, some users would
+like to configure the speed at which that dropdown happens. To support this
+functionality, the `globalSummon` action will support the following property:
+
+* `"dropdownDuration": float`
+  - When omitted, `0`, or a negative number: No animation is used
+    when summoning the window. The summoned window is focused immediately where
+    it is.
+  - When a positive number is provided, the terminal will use that value as a
+    duration (in seconds) to slide the terminal into position when activated.
+  - The default would be some sensible value. The pane animation is .2s, so
+    `0.2` might be a resonable default here.
+
+We could have alternatively provided a `"dropdownSpeed"` setting, that provided
+a number of pixels per second. In my opinion, that would be harder for users to
+use correctly. I believe that it's easier for users to mentally picture "I'd
+like the dropdown to last 100ms" vs "My monitor is 1504px tall, so I need to set
+this to 15040 to make the window traverse the entire display in .1s"
+
+Some users might want to be able to use the global hotkey to hide the window
+when the window is already visible. This would let the hotkey act as a sort of
+global toggle for the Terminal window. Others might not like that behavior, and
+just want the action to always bring the Terminal into focus, and do nothing if
+the terminal is already focused. To facilitate both these use cases, we'll add
+the following property:
+
+* `"hideWhenVisible": bool`
+  - When `true`: (_default_) When this hotkey is pressed, and the terminal
+    window is currently active, minimize the window.
+      - When `dropdownDuration` is not `0`, then the window will slide back off
+        the top at the same speed as it would come down.
+  - When `false`: When this hotkey is pressed, and the terminal window is
+    currently active, do nothing.
+
+### Quake Mode
+
+In addition to just summoning the window from anywhere, some terminals also
+support a special "quake mode" buffer or window. This window is one that closely
+emulates the console from quake:
+* It's docked to the top of the screen

```

Comment:
Future consideration: can we eventually add some customization option to dock this off the bottom instead?

Reply:
I was looking at my vertical monitor when typing the comment, so I thought it would be dumb... but then I looked at my horizontal monitor hahaha.

TLDR: yes, any side works

Reply:
Adding this to future considerations